source/tools/generateContractInterfaces.ts

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env node
+
+import * as fs from "async-file";
+import { Abi, AbiFunction, Primitive } from 'ethereum';
+import { CompilerOutputContracts, CompilerOutput } from "solc";
+import { Configuration } from './Configuration';
+
+doWork().then(() => {
+    process.exit();
+}).catch(error => {
+    console.log(error);
+    process.exit();
+});
+
+async function doWork(): Promise<void> {
+
+    const configuration = await Configuration.create();
+
+    const contractABIs: CompilerOutput = JSON.parse(await fs.readFile(configuration.contractOutputPath, "utf8"));
```

Comment:
Recommend having `generateContractInterfaces.ts` call `ContractCompiler.ts` and then pass the return value from that into `contractInterfacesTemplate`.  `ContractCompiler` will correctly do caching and compile if necessary (if cache is busted).  This avoids having to read the file off disk after writing it to disk.  It also avoids running into a problem where `compile` wasn't run first and re-generating the TS files using stale compilation output.